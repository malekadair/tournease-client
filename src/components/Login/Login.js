import React, { Component } from "react";

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: ""
    };
  }

  handleChange = event => {
    const { name, value, type, checked } = event.target;
    type === "checkbox"
      ? this.setState({ [name]: checked })
      : this.setState({ [name]: value });
  };

  render() {
    return (
      <section className="login">
        <h1>Login</h1>
        <form className="loginForm">
          <label htmlFor="username">Username:</label>
          <input
            type="text"
            name="username"
            value={this.state.username}
            required
            onChange={this.handleChange}
          />
          <br />

          <label htmlFor="password">password:</label>
          <input
            type="text"
            name="password"
            value={this.state.password}
            required
            onChange={this.handleChange}
          />
          <br />
          <button>Login</button>
        </form>
      </section>
    );
  }
}

export default Login;
