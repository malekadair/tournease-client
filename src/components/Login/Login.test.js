import React from "react";
import toJson from "enzyme-to-json";
import Login from "./Login";

describe(`Login component`, () => {
  const props = {
    id: "a",
    name: "test-class-name"
  };

  it("renders a .Login by default", () => {
    const wrapper = <Login />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the Login given props", () => {
    const wrapper = <Login {...props} />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
