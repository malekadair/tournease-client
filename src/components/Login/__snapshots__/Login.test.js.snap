// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders a .Login by default 1`] = `undefined`;

exports[`Login component renders the Login given props 1`] = `undefined`;
